### PR TITLE
Add period_days to analyze stage_time_analysis for jobs over the last so many days

### PR DESCRIPTION
**Pull request by andrew-m-leonard**
It is more useful if this tool can analyze jobs run over the last x days, as some jobs are not run anymore or regularly and are not of interest.

Signed-off-by: Andrew Leonard <anleonar@redhat.com>

### CODE REVIEW

File: tools/stage_time_analysis.groovy [hunk 1]
```diff
@@ -15,7 +15,7 @@ limitations under the License.
 import groovy.json.JsonSlurper
```
Comment (M-Davies) [inline tools/stage_time_analysis.groovy:15]: import not indented with the rest of the top level code

File: tools/stage_time_analysis.groovy [hunk 2]
```diff
@@ -27,52 +27,62 @@ import groovy.json.JsonSlurper
...
-        getJobStageDuration(statsFile, jobName, findStage)
+        getJobStageDuration(since, statsFile, jobName, findStage)
       }
     }
   }
 
 node ("master") {
   def jenkinsUrl = "${params.JENKINS_URL}"
   def findStage = "${params.STAGE}"
+  def periodDays = params.PERIOD_DAYS as Long
```
Comment (M-Davies) [inline tools/stage_time_analysis.groovy:66]: Is this ok not having a null catcher?

Comment (andrew-m-leonard) [reply tools/stage_time_analysis.groovy:66]: These are mandatory params and have config defaults

File: tools/stage_time_analysis.groovy (hunk 1 above)
Comment (andrew-m-leonard) [reply tools/stage_time_analysis.groovy]: fixed

**Review by M-Davies — APPROVED** [review]

**Review by karianna — APPROVED** [review]
lgtm

**Pull request merged**